src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        addBottom(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkRangeAdd(index);
+        if (index == 0) {
+            addTop(value);
+        } else if (index == size) {
+            addBottom(value);
+        } else {
+            addMiddle(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            addBottom(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkRange(index);
+        Node<T> nodeAtIndex = getNode(index);
+        T oldValue = nodeAtIndex.value;
+        nodeAtIndex.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkRange(index);
+        Node<T> toRemove = getNode(index);
+        unlink(toRemove);
+        return toRemove.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = first;
+        while (currentNode != null) {
+            if (areEqual(currentNode.value, object)) {
+                unlink(currentNode);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
-    @Override
-    public boolean isEmpty() {
-        return false;
+    @Override public boolean isEmpty() {
+        return size == 0;
+    }
+
+    private void checkRange(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index of element: " + index
+                    + "Size of the list: " + size);
+        }
+    }
+
+    private void checkRangeAdd(int index) {
+        if (index < 0 || index > size + 1) {
+            throw new IndexOutOfBoundsException("Index of elements: " + index
+                    + " Size of the list: " + size);
+        }
+    }
```

Comment:
I think you can remove this method as it's actually not necessary. Also remove its call from add() method